src/universal_transfer_operator/integrations/fivetran/connector/aws/s3.py

```diff
@@ -0,0 +1,292 @@
+    :param paused: Specifies whether the connector is paused. Defaults to False.
+    :param pause_after_trial: Specifies whether the connector should be paused after the free trial period has ended.
+     Defaults to False.
+    :param sync_frequency: The connector sync frequency in minutes Enum: "5" "15" "30" "60" "120" "180" "360" "480"
+     "720" "1440".
+    :param daily_sync_time: Defines the sync start time when the sync frequency is already set
+    :param schedule_type: Define the schedule type
+    :param connect_card_config: Connector card configuration
+    :param trust_certificates: Specifies whether we should trust the certificate automatically. The default value is
+     FALSE. If a certificate is not trusted automatically,
+    :param trust_fingerprints: Specifies whether we should trust the SSH fingerprint automatically. The default value
+     is FALSE.
+    :param run_setup_tests: Specifies whether the setup tests should be run automatically. The default value is TRUE.
+    """
+
+    def __init__(
+        self,
+        connector_id: str | None,
+        service: str,
+        config: dict,
+        paused: bool = False,
+        pause_after_trial: bool = False,
+        sync_frequency: str | None = None,
+        daily_sync_time: str | None = None,
+        schedule_type: str | None = None,
+        trust_certificates: bool = False,
+        trust_fingerprints: bool = False,
+        run_setup_tests: bool = True,
+        connect_card_config: dict | None = None,
+    ):
+        self.connector_id = connector_id
+        self.service = service
+        self.config = config
+        self.paused = paused
+        self.pause_after_trial = pause_after_trial
+        self.sync_frequency = sync_frequency
+        self.daily_sync_time = daily_sync_time
+        self.schedule_type = schedule_type
+        self.trust_certificates = trust_certificates
+        self.trust_fingerprints = trust_fingerprints
+        self.run_setup_tests = run_setup_tests
+        self.connect_card_config = connect_card_config
+        super().__init__(
+            connector_id=self.connector_id,
+            service=self.service,
+            config=self.config,
+            paused=self.paused,
+            pause_after_trial=self.pause_after_trial,
+            sync_frequency=self.sync_frequency,
+            daily_sync_time=self.daily_sync_time,
+            schedule_type=self.schedule_type,
+            trust_certificates=self.trust_certificates,
+            trust_fingerprints=self.trust_fingerprints,
+            run_setup_tests=self.run_setup_tests,
+            connect_card_config=self.connect_card_config,
+        )
+
+    def to_dict(self) -> dict:
+        """
+        Convert options class to dict
+        """
+        return vars(self)
+
+    def hook(self, conn_id: str, verify: Any, transfer_config_args: Any, s3_extra_args: Any) -> S3Hook:
```

Comment:
Based on your `FivetranConnector` base class definition think the interface for `S3Connector` should only contain three public methods. WDYT?

Reply:
Yes, that's correct. Other methods are just to support `map_airflow_connection_to_fivetran()` method here: https://github.com/astronomer/apache-airflow-provider-transfers/blob/fivetran-setup/src/universal_transfer_operator/integrations/fivetran/connector/aws/s3.py#L95

Reply:
So we should make other methods private right?

Reply:
[done](https://github.com/astronomer/apache-airflow-provider-transfers/pull/62/commits/7d597d98b425748d98d8778fac65423d0332fbca)